src/main/scala/com/microsoft/hyperspace/index/rules/RuleUtils.scala

```diff
@@ -381,18 +383,36 @@ object RuleUtils {
       filesAppended: Seq[Path]): LogicalPlan = {
     // Transform the location of LogicalRelation with appended files.
     val planForAppended = originalPlan transformDown {
-      case baseRelation @ LogicalRelation(fsRelation: HadoopFsRelation, baseOutput, _, _) =>
+      case baseRelation @ LogicalRelation(
+            fsRelation @ HadoopFsRelation(location: PartitioningAwareFileIndex, _, _, _, _, _),
+            baseOutput,
+            _,
+            _) =>
+        val options = if (location.partitionSchema.isEmpty) {
+          Map[String, String]()
+        } else {
+          // Retrieve basePath from a partition path.
+          val basePath = location.partitionSpec.partitionColumns
+            .foldLeft(location.partitionSpec.partitions.head.path) { (path, _) =>
+              path.getParent
+            }
+
+          // Set "basePath" so that partitioned columns are also included in the output schema.
+          Map("basePath" -> basePath.toString)
```

Comment:
could you please point me to where this key-value is being used. I couldn't find it.

Reply:
It's used in https://github.com/apache/spark/blob/master/sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/PartitioningAwareFileIndex.scala#L221.